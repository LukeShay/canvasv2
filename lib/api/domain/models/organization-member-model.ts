/* eslint-disable @typescript-eslint/no-var-requires */
/* eslint-disable global-require */
import { Model, JSONSchema, RelationMappings } from 'objection';
import Constants from '../constants';

export class OrganizationMemberModel extends Model {
  static tableName = Constants.ASSISTANTS_TABLE;

  static idColumn = 'id';

  static get jsonSchema(): JSONSchema {
    return {
      properties: {
        organizationId: { type: 'string' },
        role: { type: 'string' },
        userId: { type: 'string' },
      },
      required: ['organizationId', 'role', 'userId'],
      type: 'object',
    };
  }

  static get relationMappings(): RelationMappings {
    // ts-expect-error
    const { UserModel } = require('./user-model');
    // ts-expect-error
    const { OrganizationModel } = require('./organization-model');

    return {
      organization: {
        join: {
          from: `${OrganizationMemberModel.tableName}.organizationId`,
          to: `${OrganizationModel.tableName}.${OrganizationModel.idColumn}`,
        },
        modelClass: OrganizationModel,
        relation: Model.BelongsToOneRelation,
      },
      user: {
        join: {
          from: `${OrganizationMemberModel.tableName}.userId`,
          to: `${UserModel.tableName}.${UserModel.idColumn}`,
        },
        modelClass: UserModel,
        relation: Model.BelongsToOneRelation,
      },
    };
  }
}
