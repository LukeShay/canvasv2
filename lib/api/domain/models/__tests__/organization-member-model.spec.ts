import { OrganizationMemberModel } from '../organization-member-model';

describe('OrganizationMemberModel', () => {
  it("should have idColum 'id'", () => {
    expect(OrganizationMemberModel.idColumn).toEqual('id');
  });

  it('should should have jsonSchema that matches snapshot', () => {
    expect(OrganizationMemberModel.jsonSchema).toMatchSnapshot();
  });

  it('should should have relationMappings that matches snapshot', () => {
    expect(OrganizationMemberModel.relationMappings).toMatchSnapshot();
  });
});
