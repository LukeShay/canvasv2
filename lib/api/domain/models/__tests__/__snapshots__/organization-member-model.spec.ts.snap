// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizationMemberModel should should have jsonSchema that matches snapshot 1`] = `
Object {
  "properties": Object {
    "organizationId": Object {
      "type": "string",
    },
    "role": Object {
      "type": "string",
    },
    "userId": Object {
      "type": "string",
    },
  },
  "required": Array [
    "organizationId",
    "role",
    "userId",
  ],
  "type": "object",
}
`;

exports[`OrganizationMemberModel should should have relationMappings that matches snapshot 1`] = `
Object {
  "organization": Object {
    "join": Object {
      "from": "Assistants.organizationId",
      "to": "Assistants.id",
    },
    "modelClass": [Function],
    "relation": [Function],
  },
  "user": Object {
    "join": Object {
      "from": "Assistants.userId",
      "to": "Users.id",
    },
    "modelClass": [Function],
    "relation": [Function],
  },
}
`;
